jsf/WFLY-13350_galleon_layer_for_jsf_support.adoc

```diff
@@ -0,0 +1,102 @@
+= Galleon layer for Jakarta Server Faces support
+:author:            Yeray Borges
+:email:             yborgess@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Jakarta Server Faces (JSF) defines an MVC framework for building user interfaces for web applications, including UI components, state management, event handing, input validation, page navigation, and support for internationalization and accessibility.
+
+The goal of this feature is to supply a https://docs.wildfly.org/galleon/#_layers[Galleon layer] which will be responsible for providing the WildFly JSF subsystem.
+
+The following is a configuration example of the WildFly bootable JAR Maven plugin to provision a server with the `jsf` layer:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-jar-maven-plugin</artifactId>
+    <configuration>
+        <feature-pack-location>wildfly@maven(org.jboss.universe:community-universe)#${version.wildfly}</feature-pack-location>
+        <layers>
+            <layer>jsf</layer>
+        </layers>
+    </configuration>
+    <executions>
+        <execution>
+            <goals>
+                <goal>package</goal>
+            </goals>
+        </execution>
+    </executions>
+</plugin>
+----
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13350[WFLY-13350 Add a Galleon layer for JSF]
+* https://issues.redhat.com/browse/EAP7-1529[EAP7-1529 Add a Galleon layer for JSF]
+
+=== Related Issues
+
+=== Dev Contacts
+
+* mailto:brian.stansberry@redhat.com[Brian Stansberry]
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+* The Galleon layer must provide a servlet container and configure the WildFly JSF subsystem.
+
```

Comment:
@tommaso-borgato The JSF layer doesn't need to depend on the EJB3 layer. JSF is an independent layer and it does not need the presence of an EJB3 container to work, obviously if you plan to use an EJB with your JSF managed beans, you will need to provision one of the EJB3 flavours, depending on what you need.

Your finding about the module dependencies is a good, I suspect the JSF implementation uses some of the interfaces available on the `javax.ejb.api`. We can ignore it, because Galleon will provision `javax.ejb.api` together with the `com.sun.jsf-impl`.